frappe/email/doctype/email_account/email_account.js

```diff
@@ -145,32 +145,31 @@ frappe.ui.form.on("Email Account", {
 					"email_id": cur_frm.doc.email_id
 				},
 				callback: function (frm) {
-					try {
-						if (cur_frm.doc.domain !=frm["message"][0]["name"]) {
+					if (frm.message) {
+						if (cur_frm.doc.domain != frm["message"][0]["name"]) {
 							cur_frm.doc.domain = frm["message"][0]["name"]
-							cur_frm.doc.email_server= frm["message"][0]["email_server"];
-							cur_frm.doc.use_imap= frm["message"][0]["use_imap"];
-							cur_frm.doc.smtp_server= frm["message"][0]["smtp_server"];
-							cur_frm.doc.use_ssl= frm["message"][0]["use_ssl"];
-							cur_frm.doc.use_tls= frm["message"][0]["use_tls"];
+							cur_frm.doc.email_server = frm["message"][0]["email_server"];
+							cur_frm.doc.use_imap = frm["message"][0]["use_imap"];
+							cur_frm.doc.smtp_server = frm["message"][0]["smtp_server"];
+							cur_frm.doc.use_ssl = frm["message"][0]["use_ssl"];
+							cur_frm.doc.use_tls = frm["message"][0]["use_tls"];
 							cur_frm.doc.smtp_port = frm["message"][0]["smtp_port"];
 							if (!norefresh) {
 								cur_frm.refresh();
 							}
 						}
-					}
-					catch (Exception) {
+					}else{
 						frappe.confirm(
-							'Email Domain not configured for this account\nCreate one?',
-							function () {
-								frappe.model.with_doctype("Email Domain", function() {
-									frappe.route_options = {email_id: cur_frm.doc.email_id};
-									frappe.route_titles["return to email_account"] = 1
-									var doc = frappe.model.get_new_doc("Email Domain");
-									frappe.set_route("Form", "Email Domain", doc.name);
-								})
-							}
-						)
+						'Email Domain not configured for this account\nCreate one?',
```

Comment:
Messages should be translatable `__("Email...")`

also, don't use `\n` in messages, messes up translations. Comma would suffice here.